### PR TITLE
Switch to dart-sass

### PR DESCRIPTION
**Pull request by XhmikosR**
This is something we've been discussing internally for some time now.

Things to note:

1. dart-sass's precision is hardcoded to 10; I dropped it down to 6 for the minified files, but we might want to reconsider this and just go with the default? I need to check how much it adds, though
2. dart-sass orders grouped selectors differently; I have not found a way to work around this and perhaps we should ask upstream. Do note that the minified files keep the old behavior since clean-css sorts the grouped selectors alphabetically
3. dart-sass converts escaped UTF-8 characters which in turn means we need a `@charset "UTF-8";`. Not the end of the world, but it is something to note. Also, note that there's a `--no-charset` flag, but then this could be dangerous since we do have Unicode characters, so we better not use it
4. dart-sass removes quotes from attribute selectors; not a big deal since they are already removed in the minified files
5. dart-sass adds leading zeros; again, not a big deal since they are already removed in the minified files
6. other minor output differences like whitespaces

@twbs/css-review for discussion

----

1. Check the bundlewatch size differences
2. Live preview: <https://deploy-preview-31728--twbs-bootstrap.netlify.app/>
3. Test that sourcemaps work properly; I see some changes in soucemaps' `sources` property

Refs #29853

### CODE REVIEW

File: dist/css/bootstrap.css [hunk 1]
```diff
@@ -2320,42 +2468,35 @@ progress {
...
-
 .form-file-lg .form-file-text,
 .form-file-lg .form-file-button {
   padding: 0.5rem 1rem;
 }
 
 .form-range {
   width: 100%;
-  height: 1.4rem;
+  height: calc(1rem + 0.5em);
```
Comment (MartijnCuppens) [inline dist/css/bootstrap.css:2478]: Caused by https://github.com/twbs/bootstrap/commit/1af79bbc5267428af9fcb4a8fc9e6ab0b30328f4. We might need to look into the other effects of the unit change here (not related to this PR btw)

Comment (XhmikosR) [reply dist/css/bootstrap.css:2478]: Yeah, that's why I decided to split the dist commits so that we can spot the dart-sass only changes.

Comment (MartijnCuppens) [reply dist/css/bootstrap.css:2478]: Opened https://github.com/twbs/bootstrap/pull/31746 to fix the unit

**Review by mdo — APPROVED** [review]
Pre-approving for when this comes up for review.

**Comment by XhmikosR** [comment]
Let's land it as is and we can always revisit the clean-css options later. BTW we could also ask upstream for the output discrepancies and if they can do something about it, like with sorting grouped selectors.

**Pull request merged**